### ISSUE TITLE
Exception: A sheet with the name ‘#######@gmail.com’ already exists. Please enter another name.

### ISSUE THREAD

**Issue by EdnarStuck** [fleapower/Google-Script-Contacts-Sync#2]
Hi! I can't get past this error when trying to sync 2 accounts.
Just to get the checklist off:
- I have created and shared the project script from my main to my client account
- Every "email1/2@gmail.com" from the first part of the script (the tutorial) is replaced with the main and client account
- There's a blank spreadsheet shared from the main to the client account (the ID is copied on the script)
- I have created and uploaded a blank .txt file with the name '#####@gmail.com_PeopleSyncToken.txt' to both accounts (not the same document uploaded to main and shared to the client account, but one single file uploaded to both accounts)
- People api is loaded onto the main account and the 'MasterInit' and the ERROR 1 shown is from running it
- ERROR 2 is from the client account. I'm posting it just in case it may help resolve this

ERROR 1 (Main acc.)
Notice   Execution started
Info	       #####@gmail.com
Info	       getContactGroupsList
Error       Exception: A sheet with the name ‘jipf98@gmail.com’ already exists. Please enter another name.
               MasterInit	  @ Code.gs:118

ERROR 2 (Client acc.)
Notice	Execution started
Info	       #######@gmail.com
Info	        getContactGroupsList
Info	        deleteAllTriggers
Info	        RefreshSynctoken
Error	Exception: Cannot retrieve the next object: iterator has reached the end.
                RefreshSyncToken	@ Code.gs:523
                ClientInit	                @ Code.gs:201

Thank you! If there's anything else you need to know reply me and I'll get it to you

**Comment by fleapower** [comment]
Sorry for the issues.  It is strange that it is throwing this error since I had another user have the exact opposite problem.  I'm away from my desk until Monday and am in the mountains with very little coverage.  In other words, I can't fix the script until Monday.  If you can't await, try deleting line 118 and see if that works.  If you can wait until Monday, I'll work on it then.  I'm pretty sure I know what the problem is (bad error handling and/or wrong instructions).

**Comment by EdnarStuck** [comment]
No worries! I'll wait. Thank you.

**Comment by fleapower** [comment]
Hmm.  Don't try running the ClientInit until you get a clean run on MasterInit.

An error at line 531 could be lots of things.  Did you change anything else in the code other than the email addresses and file ID for the spreadsheet?

Try inserting the following code before line 531:

`Logger.log("Sync Token: " + syncToken);`

The code with the lines before and after should look like this:

```
    if (!syncTokenExpired) {
      Logger.log("Sync Token: " + syncToken);
      var connections = People.People.Connections.list('people/me', {
```

Post the log results here.

**Comment by EdnarStuck** [comment]
Okay, I won't try with the client account yet.
I haven't changed anything prior line 51. The only things I've changed are the 'email1/2@gmail.com' and spreadsheet ID.

The log results have improved (I think haha):
```
8:18:28 PM	Notice	Execution started
8:18:29 PM	Info	jipf98@gmail.com
8:18:29 PM	Info	getContactGroupsList
8:18:30 PM	Info	RefreshSynctoken
8:18:30 PM	Info	Sync Token: %PDF-1.4
                        %����
                        1 0 obj
                        <</Title (jipf98@gmail.com_PeopleSyncToken.txt)
                        /Producer (Skia/PDF m103 Google Docs Renderer)>>
                        endobj
                        3 0 obj
                        <</ca 1
                        /BM /Normal>>
                        endobj
                        4 0 obj
                        <</Length 84>> stream
                        1 0 0 -1 0 792 cm
                        q
                       .75 0 0 .75 0 0 cm
                        1 1 1 RG 1 1 1 rg
                        /G3 gs
                       0 0 816 1056 re
                       f
                       Q
 
                       endstream
                       endobj
                       2 0 obj
                       <</Type /Page
                       /Resources <</ProcSet [/PDF /Text /ImageB /ImageC /ImageI]
                       /ExtGState <</G3 3 0 R>>>>
                       /MediaBox [0 0 612 792]
                       /Contents 4 0 R
                       /StructParents 0
                       /Parent 5 0 R>>
                       endobj
                       5 0 obj
                       <</Type /Pages
                       /Count 1
                       /Kids [2 0 R]>>
                       endobj
                       6 0 obj
                       <</Type /Catalog
                       /Pages 5 0 R>>
                       endobj
                       xref
                       0 7
                       0000000000 65535 f 
                       0000000015 00000 n 
                       0000000296 00000 n 
                       0000000127 00000 n 
                       0000000164 00000 n 
                       0000000484 00000 n 
                       0000000539 00000 n 
                       trailer
                       <</Size 7
                       /Root 6 0 R
                       /Info 1 0 R>>
                       startxref
                       586
                       %EOF
8:18:30 PM	Error  GoogleJsonResponseException: API call to people.people.connections.list failed with error: Request contains an invalid argument.
                       RefreshSyncToken	@ Code.gs:532
                       MasterInit	 @ Code.gs:122
```

**Comment by fleapower** [comment]
It looks like your synctoken file is corrupted.  Try deleting it and creating a new one.  Be sure to use a basic text editor on your computer and then upload it to to Google Drive.  Don't create it using Google Docs.

**Comment by EdnarStuck** [comment]
I've deleted the previous .txt and created a new one with this name: '#####@gmail.com_PeopleSyncToken.txt', uploaded to the main and client Google Drive accounts. Refreshed the Google Contacts Script page and ran it again (MasterInit, on main account).

Same error persists.

```
Error	GoogleJsonResponseException: API call to people.people.connections.list failed with error: Request contains an invalid argument.
        RefreshSyncToken @ Code.gs:532
        MasterInit @ Code.gs:122
```

**Comment by fleapower** [comment]
Just to be clear, you are using your email address for the beginning of the text file, correct?  Something like

myname@gmail.com_PeopleSyncToken.txt

What was the output when the Sync Token was logged that time?

**Comment by EdnarStuck** [comment]
Yes, the file is named just as you're saying (with my info, of course).

I'm not following on the output of the Sync Token. How can I see this?

**Comment by fleapower** [comment]
In the output log when you run the script, the line you added logs the current sync token.  It is going to show up as "Sync Token:" followed by the current sync token value.  It is going to be a long string of characters.

**Comment by EdnarStuck** [comment]
Found it (if I'm not mistaken)! It's the same as the one in the comment earlier:

```
Sync Token: %PDF-1.4
%����
1 0 obj
<</Title (#####@gmail.com_PeopleSyncToken.txt)
/Producer (Skia/PDF m103 Google Docs Renderer)>>
endobj
3 0 obj
<</ca 1
/BM /Normal>>
endobj
4 0 obj
<</Length 84>> stream
1 0 0 -1 0 792 cm
q
.75 0 0 .75 0 0 cm
1 1 1 RG 1 1 1 rg
/G3 gs
0 0 816 1056 re
f
Q

endstream
endobj
2 0 obj
<</Type /Page
/Resources <</ProcSet [/PDF /Text /ImageB /ImageC /ImageI]
/ExtGState <</G3 3 0 R>>>>
/MediaBox [0 0 612 792]
/Contents 4 0 R
/StructParents 0
/Parent 5 0 R>>
endobj
5 0 obj
<</Type /Pages
/Count 1
/Kids [2 0 R]>>
endobj
6 0 obj
<</Type /Catalog
/Pages 5 0 R>>
endobj
xref
0 7
0000000000 65535 f 
0000000015 00000 n 
0000000296 00000 n 
0000000127 00000 n 
0000000164 00000 n 
0000000484 00000 n 
0000000539 00000 n 
trailer
<</Size 7
/Root 6 0 R
/Info 1 0 R>>
startxref
586
%EOF
```

**Comment by fleapower** [comment]
Can you open the sync token file in Google Drive and see what it contains?

**Comment by EdnarStuck** [comment]
It's a blank Google Docs file

**Comment by fleapower** [comment]
Are you using Windows?  Would you be agreeable to using Quick Assist so I can try to see what is going on?

**Comment by EdnarStuck** [comment]
Okay. We can try.

**Comment by fleapower** [comment]
Sorry...got really busy.  I have some time today and tomorrow if you want to try to fix this.

**Comment by EdnarStuck** [comment]
No worries. How can we follow this up privately?

**Comment by fleapower** [comment]
Can you PM me on Reddit?  Same username.

**Comment by EdnarStuck** [comment]
Okay, a little update if someone gets the same error as I did. It was all just pretty simple and I hadn't noticed until @fleapower told me.
When uploading the token (.txt file "emailX@gmail.com_PeopleSyncToken.txt") into Drive, I had the option enabled to convert any file into a Google Docs if necessary. With that turned off, the tokens worked just fine and everything is now synched.

Also, a few things I noticed I was doing wrong, too. When uploading this .txt, sometimes I'd get the ending of the file name like this ".txt.txt". Just delete one of these and you're good to go.

Thank you again @fleapower for your patience and for making this available to anyone.

**Comment by fleapower** [comment]
Glad it's finally working!  I will update the instructions.
